### PR TITLE
[Feature] Trino dialect, the extract function supports dow and week parameters

### LINKED ISSUE #52698: Trino dialect the extract function not supports dow and week parameters
- [x] https://github.com/StarRocks/starrocks/pull/52651

### Steps to reproduce the behavior (Required)
set sql_dialect='trino';
SELECT extract(dow FROM TIMESTAMP '2022-10-20 05:10:00');

### Expected behavior (Required)
输出结果值 4
### Real behavior (Required)
返回异常 Getting analyzing error. Detail message: No matching function with signature: dow(datetime)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
3.3.5

### PR DESCRIPTION
**Pull request by renzhimin7**
## Why I'm doing:
```
extract(dow FROM TIMESTAMP '2022-10-20 05:10:00') not support trino dialect.

set sql_dialect='trino';
SELECT extract(dow FROM TIMESTAMP '2022-10-20 05:10:00');
ERROR 1064 (HY000): Getting analyzing error. Detail message: No matching function with signature: dow(datetime).

```
## What I'm doing:

Fixes #52698 

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
  - [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by miomiocat — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java [hunk 1]
```diff
@@ -999,6 +999,11 @@ protected ParseNode visitCoalesceExpression(CoalesceExpression node, ParseTreeCo
     @Override
     protected ParseNode visitExtract(Extract node, ParseTreeContext context) {
         String fieldString = node.getField().toString();
+        if (fieldString.equalsIgnoreCase("dow")) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java:1002]: you can use `Trino2SRFunctionCallTransformer.convert(node.getName().toString(), arguments)`,  no need transform function here

Comment (renzhimin7) [reply fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java:1002]: I don't quite understand, can you describe it in detail @Youngwb

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java:1002]:
```
protected ParseNode visitExtract(Extract node, ParseTreeContext context) {
        String fieldString = node.getField().toString().toLowerCase();
        return  Trino2SRFunctionCallTransformer.convert(fieldString,
                Lists.newArrayList((Expr) visit(node.getExpression(), context)));
    }
```

Comment (renzhimin7) [reply fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/AstBuilder.java]: ths

**Review by Youngwb — DISMISSED** [review]

**Review by trueeyu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java [hunk 2]
```diff
@@ -232,6 +232,8 @@ private static void registerDateFunctionTransformer() {
         // last_day_of_month(x)  -> last_day(x,'month')
         registerFunctionTransformer("last_day_of_month", 1, new FunctionCallExpr("last_day",
                 List.of(new PlaceholderExpr(1, Expr.class), new StringLiteral("month"))));
+        // year  -> year
```
Comment (dirtysalt) [inline fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java:235]: why have this year->year function?

File: fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java [hunk 3]
```diff
@@ -232,6 +232,11 @@ private static void registerDateFunctionTransformer() {
         // last_day_of_month(x)  -> last_day(x,'month')
         registerFunctionTransformer("last_day_of_month", 1, new FunctionCallExpr("last_day",
                 List.of(new PlaceholderExpr(1, Expr.class), new StringLiteral("month"))));
+
+        // year_of_week  -> weekofyear
+        registerFunctionTransformer("year_of_week", 1, "weekofyear", List.of(Expr.class));
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java:237]: these function can transform? they have different semantic

**Review by Youngwb — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**